Tests/AK/TestHashFunctions.cpp

```diff
@@ -53,3 +54,53 @@ TEST_CASE(constexpr_ptr_hash)
     // "ptr_hash" test binds the result.
     static_assert(ptr_hash(FlatPtr(42)));
 }
+
+// Testing concrete hash results is not possible due to SipHash's non-determinism.
+// We instead perform some sanity checks and try to hit any asserts caused by programming errors.
+TEST_CASE(sip_hash)
+{
+    static SipHash<1, 3> StandardSipHash;
+    // default as used in the reference implementation
+    static SipHash<2, 4> UpstreamDefaultSipHash;
+    static SipHash<4, 8> SecureSipHash;
+
+    EXPECT_EQ(StandardSipHash(42), StandardSipHash(42));
+    EXPECT_EQ(UpstreamDefaultSipHash(42), UpstreamDefaultSipHash(42));
+    EXPECT_EQ(SecureSipHash(42), SecureSipHash(42));
+    EXPECT_NE(StandardSipHash(42), UpstreamDefaultSipHash(42));
+    EXPECT_NE(StandardSipHash(42), SecureSipHash(42));
+}
+
+TEST_CASE(sip_hash_bytes)
+{
+    constexpr Array<u8, 8> short_test_array { 1, 2, 3, 4, 5, 6, 7, 8 };
+    constexpr Array<u8, 16> common_prefix_array { 1, 2, 3, 4, 5, 6, 7, 8, 0, 0, 0, 0, 0, 0, 0, 0 };
+    EXPECT_EQ((sip_hash_bytes<1, 3>(short_test_array.span())), (sip_hash_bytes<1, 3>(short_test_array.span())));
+    EXPECT_NE((sip_hash_bytes<1, 3>(short_test_array.span())), (sip_hash_bytes<1, 3>(common_prefix_array.span())));
+
+    for (size_t prefix_length = 1; prefix_length < 8; ++prefix_length) {
+        EXPECT_NE((sip_hash_bytes<1, 3>(short_test_array.span().trim(prefix_length))), (sip_hash_bytes<1, 3>(short_test_array.span())));
+        EXPECT_EQ((sip_hash_bytes<1, 3>(short_test_array.span().trim(prefix_length))), (sip_hash_bytes<1, 3>(common_prefix_array.span().trim(prefix_length))));
+    }
+}
+
+static void run_benchmark(Function<unsigned(u64)> hash_function)
+{
+    for (size_t i = 0; i < 1'000'000; ++i)
+        EXPECT_EQ(hash_function(i), hash_function(i));
+}
```

Comment:
This is a _very_ bad benchmark, the overhead of calling `Function::operator()` is comparable to the actual time spent hashing a u64.
And sweeping over 0..1000000 for a hash benchmark is...not very realistic (though it's OK for a tiny microbenchmark)

Reply:
Of course this benchmark has to run multiple times when you actually want to check performance. A single benchmark run should be <1s so it doesn't bog down CI; 1 million fits nicely with that and you can run the benchmark 1000 times or so to get a good averaged measurement; it still runs often enough within one benchmark to filter out overhead.

Reply:
we don't run benchmarks in CI